fe/fe-core/src/main/java/com/starrocks/catalog/EsResource.java

```diff
@@ -0,0 +1,126 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.catalog;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.gson.annotations.SerializedName;
+import com.starrocks.common.DdlException;
+import com.starrocks.common.proc.BaseProcResult;
+import com.starrocks.connector.elasticsearch.external.EsUtil;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Map;
+
+import static com.starrocks.catalog.EsTable.TRANSPORT;
+import static com.starrocks.catalog.EsTable.TRANSPORT_HTTP;
+import static com.starrocks.catalog.EsTable.TRANSPORT_THRIFT;
+
+/**
+ * ES resource
+ * <p>
+ * Syntax:
+ * CREATE RESOURCE "remote_es"
+ * PROPERTIES
+ * (
+ * "type" = "es",
+ * "hosts" = "http://192.168.0.1:8200,http://192.168.0.2:8200",
+ * "index" = "test",
+ * "es.type" = "doc",
+ * "user" = "root",
+ * "password" = "root"
+ * );
+ */
+public class EsResource extends Resource {
```

Comment:
There is no need to create an EsResource  for compatibility.  Because later we mainly recommend users to use catalog rather than resource.